webapp/src/components/CreateEstatePage/EstateSelect/EstateSelect.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Container } from 'semantic-ui-react'
+
+import ParcelPreview from 'components/ParcelPreview'
+import EstateSelectActions from './EstateSelectActions'
+import Parcel from 'components/Parcel'
+import { t } from 'modules/translation/utils'
+import { isOwner } from 'modules/parcels/utils'
+import { coordsType } from 'components/types'
+
+import './EstateSelect.css'
+
+export default class EstateSelect extends React.PureComponent {
+  static propTypes = {
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+    error: PropTypes.string,
+    onCancel: PropTypes.func.isRequired,
+    onContinue: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(coordsType).isRequired
+  }
+
+  isNeighbour = (x, y) => coords => {
+    return (
+      (coords.x === x && (coords.y + 1 === y || coords.y - 1 === y)) ||
+      (coords.y === y && (coords.x + 1 === x || coords.x - 1 === x))
+    )
+  }
+
+  hasNeighbour = (x, y) => {
+    const { value: parcels } = this.props
+    return parcels.some(this.isNeighbour(x, y))
+  }
+
+  getNeighbours = (x, y, parcels = this.props.parcels) => {
+    return parcels.filter(this.isNeighbour(x, y))
+  }
+
+  areConnected = (parcels, alreadyTraveled = [], remaining = [...parcels]) => {
+    if (alreadyTraveled.length === parcels.length) {
+      return true
+    }
+
+    if (remaining.length === 0) {
+      return false
+    }
+
+    let actual = remaining.pop()
+
+    const neighbours = this.getNeighbours(actual.x, actual.y, parcels).filter(
+      coords => {
+        return (
+          parcels.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          ) &&
+          !alreadyTraveled.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          )
+        )
+      }
+    )
+
+    return this.areConnected(
+      parcels,
+      [...alreadyTraveled, ...neighbours],
+      remaining
+    )
+  }
+
+  handleParcelClick = wallet => (x, y) => {
+    if (!isOwner(wallet, x, y)) {
+      return
+    }
+
+    if (!this.hasNeighbour(x, y)) {
+      return
+    }
+
+    const { value: parcels, onChange } = this.props
+    const isSelected = parcels.some(coords => coords.x === x && coords.y === y)
+
+    if (isSelected) {
+      const newParcels = parcels.filter(
+        coords => !(coords.x === x && coords.y === y)
```

Comment:
I think this Component might benefit from having a method that compares parcels somewhere, to stop repeating this logic: `coords.x === x && coords.y === y` and to be able to negate: `!isEqual(parcel1, parcel2)` clearly

There are a few more places where a check like is performed like Parcel.container.js and ParcelCanvas.js